src/pool/srv_pool_check.c

```diff
@@ -307,3 +307,417 @@ ds_pool_clues_print(struct ds_pool_clues *clues)
 			dc->bcl_oid_next);
 	}
 }
+
+int
+ds_pool_clues_find_rank(struct ds_pool_clues *clues, d_rank_t rank)
+{
+	int i;
+
+	for (i = 0; i < clues->pcs_len; i++)
+		if (clues->pcs_array[i].pc_rank == rank)
+			return i;
+	return -DER_NONEXIST;
+}
+
+/*
+ * Return
+ *
+ *   > 0	<x_last_term, x_last_index> is newer than <y_last_term, y_last_index>
+ *   < 0	<x_last_term, x_last_index> is older than <y_last_term, y_last_index>
+ *   = 0	<x_last_term, x_last_index> is equal to   <y_last_term, y_last_index>
+ */
+static int
+compare_logs(uint64_t x_last_term, uint64_t x_last_index,
+	     uint64_t y_last_term, uint64_t y_last_index)
+{
+	if (x_last_term > y_last_term)
+		return 1;
+	if (x_last_term < y_last_term)
+		return -1;
+
+	if (x_last_index > y_last_index)
+		return 1;
+	if (x_last_index < y_last_index)
+		return -1;
+
+	return 0;
+}
+
+/**
+ * Analyze \a clues, which must be nonempty and comprise clues about replicas
+ * of one PS, and report if this PS requires catastrophic recovery or not.
+ *
+ * \param[in]	clues		pool clues
+ * \param[out]	advice_out	when the return value is 0, the index of the
+ *				advised replica in \a clues to rebootstrap the
+ *				PS from
+ *
+ * \return	0	this PS does not require catastrophic recovery
```

Comment:
Just trying to interpret the function documentation: when return is 0, and advice_out is populated with an index, is the index value intended to be more informative than something requiring special action to recover?

Reply:
Would suggest changing only if there is some other need to update the patch. Should the ">0" documentation below describing the return value scenarios be changed to "1" to agree with the advice_out parameter documentation above (or vice versa)?

Reply:
Errrrhhh, should have written "> 0" instead of "1" for advice_out. :( I'll fix this in a later patch (if not in the current one).